web/libs/datamanager/src/utils/api-proxy/index.js

```diff
@@ -155,29 +156,32 @@ export class APIProxy {
       let responseMeta;
       const alwaysExpectJSON = options?.alwaysExpectJSON === undefined ? true : options.alwaysExpectJSON;
 
-      let shouldUseQueryCache = false;
+      /**
+       * Object to be used to control the cache for the request
+       * @type {Object}
+       * @property {number} [staleTime]
+       * @property {string} [keyPrefix]
+       */
+      let queryCacheParams = null;
       try {
         const finalParams = {
           ...(methodSettings.params ?? {}),
           ...(urlParams ?? {}),
           ...(this.sharedParams ?? {}),
         };
 
-        if (finalParams.__useQueryCache && methodSettings.queryCache) {
-          shouldUseQueryCache = true;
+        const { __useQueryCache, ...paramsForRequest } = finalParams;
 
-          const cachedData = methodSettings.queryCache(finalParams);
-
-          if (cachedData) {
-            return cachedData;
-          }
-
-          delete finalParams.__useQueryCache;
+        if (__useQueryCache) {
+          queryCacheParams = {
+            staleTime: __useQueryCache.staleTime ?? undefined,
+            keyPrefix: __useQueryCache.prefixKey ?? methodSettings.path,
```

Comment:
is there a chance `__useQueryCache` or `methodSettings` would be undefined? that could cause an expected error

Reply:
`methodSettings` no, as it is defined beforehand manually, `__useQueryCache` yes, which is why the usage of this cache depends on there being a defined `__userQueryCache` property. If it is falsey, it will just retain the existing behaviour of the call as before.